src/server/scripts/Shadowlands/zone_exiles_reach.cpp

```diff
@@ -0,0 +1,388 @@
+
+    void OnQuestStatusChange(Player* player, Quest const* /*quest*/, QuestStatus /*oldStatus*/ , QuestStatus newStatus) override
+    {
+        switch (newStatus)
+        {
+            case QUEST_STATE_COMPLETE:
+                Conversation::CreateConversation(CONVERSATION_WARMING_UP, player, *player, player->GetGUID(), nullptr);
+                break;
+            case QUEST_STATUS_REWARDED:
+                if (Creature* breka1 = player->FindNearestCreature(NPC_WARLORD_BREKA_GRIMAXE1, 5.0f))
+                    if (Creature* breka2 = player->FindNearestCreature(NPC_WARLORD_BREKA_GRIMAXE2, 75.0f))
+                        breka2->SummonPersonalClone(breka1->GetPosition(), TempSummonType(TEMPSUMMON_TIMED_DESPAWN), 18s, 0, 0, player);
+                break;
+            default:
+                break;
+        }
+    }
+};
+
+enum HordeSparingPartner
+{
+    CONVERSATION_PREFIGHT       = 14422,
+    CONVERSATION_AGGRO          = 14423,
+    CONVERSATION_JUMP           = 14424,
+    EQUIPMENT_AXE               = 175161,
+    EVENT_MOVE_TO_A_POSITION    = 1,
+    EVENT_PREFIGHT_CONVERSATION = 2,
+    EVENT_WALK_BACK             = 3,
+    NPC_SPARING_PARTNER         = 166814,
+    NPC_SPAR_POINT_ADVERTISMENT = 174971,
+    NPC_GRUNT_THROG             = 166583,
+    QUEST_KILL_CREDIT           = 155607,
+    SPELL_JUMP_BEHIND           = 312757,
+    SPELL_COMBAT_TRAINING       = 323071,
+    SPELL_SUMMON_THROG          = 325107,
+    SPELL_UPDATE_PHASE_SHIFT    = 82238,
+    SAY_I_CONCEDE               = 0
+};
+
+class conversation_horde_sparing_partner : public ConversationScript
+{
+public:
+    conversation_horde_sparing_partner() : ConversationScript("conversation_horde_sparing_partner") { }
+
+    void OnConversationCreate(Conversation* conversation, Unit* creator) override
+    {
+        std::list<Creature*> spar;
+        creator->GetCreatureListWithEntryInGrid(spar, NPC_SPARING_PARTNER, 25.0f);
+        for (Creature* creature : spar)
+        {
+            if (creature->GetDemonCreatorGUID() == creator->GetGUID()) // @TODO requires research about DemonCreator usage for summons to work
+                conversation->AddActor(creature->GetEntry(), 1, creature->GetGUID());
+        }
+    }
+};
+
+struct npc_horde_sparring_partner : public ScriptedAI
+{
+    npc_horde_sparring_partner(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        SetEquipmentSlots(false, EQUIPMENT_AXE, EQUIP_NO_CHANGE, EQUIP_NO_CHANGE);
+    }
+
+    void IsSummonedBy(WorldObject* summonerWO) override
+    {
+        _jumped = false;
+
+        Unit* summoner = summonerWO->ToUnit();
+        if (!summoner)
+            return;
+
+        if (Creature* throg = me->FindNearestCreature(NPC_GRUNT_THROG, 5.0f))
+            _throgLocation = throg->GetPosition();
+
+        if (summoner->IsPlayer())
+        {  
+            _playerGUID = summoner->GetGUID();
```

Comment:
Cant you get the summoner via me->GetSummoner() ??

Reply:
Creature has no member "GetSummoner" error message.

Reply:
but me->ToTempSummon()->GetSummoner does work